pointcloud/octree/octree.go

```diff
@@ -0,0 +1,57 @@
+// Package octree implements a octree representation of pointclouds for easy traversal and storage of
+// probability and color data
+package octree
+
+import (
+	"github.com/golang/geo/r3"
+
+	pc "go.viam.com/rdk/pointcloud"
+)
+
+// Each node in the octree is either an internal node which links to other nodes, is an empty node with
+// no points or further links, or is an occupied node which contains a single point of data that
+// includes, location, color and probability information.
+const (
+	InternalNode = NodeType(iota)
+	LeafNodeEmpty
+	LeafNodeFilled
+	octreeVersion = 1.0
+)
+
+// NodeType represents the types of nodes possible in an octree.
+type NodeType uint8
+
+// Octree is a data structure that recursively partitions 3D space into octants to represent occupancy.
+// Each node is either a internal node, empty node or child node. This implementation of an octree is
```

Comment:
a internal node --> an internal node ?

Reply:
Can you add this description from the tech spec into here? 

> An Octree is a PointCloud that is stored in an octree format for searchability and serializability.

I found this sentence helpful in understanding the Octree interface.